services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/factory/HakemusBuilder.kt

```diff
@@ -47,51 +50,91 @@ data class HakemusBuilder(
     fun withStatus(
         status: ApplicationStatus = ApplicationStatus.PENDING,
         alluId: Int = 1,
-        identifier: String = "JS000$alluId"
+        identifier: String = "JS000$alluId",
     ): HakemusBuilder {
-        applicationEntity =
-            applicationEntity.copy(
-                alluid = alluId,
-                alluStatus = status,
-                applicationIdentifier = identifier
-            )
+        applicationEntity.apply {
+            alluid = alluId
+            alluStatus = status
+            applicationIdentifier = identifier
+        }
         return this
     }
 
     fun inHandling(alluId: Int = 1) = withStatus(ApplicationStatus.HANDLING, alluId)
 
-    fun withName(name: String): HakemusBuilder = apply {
-        when (applicationEntity.applicationType) {
-            ApplicationType.CABLE_REPORT -> {
-                onCableReport { copy(name = name) }
-            }
-            ApplicationType.EXCAVATION_NOTIFICATION -> {
-                onExcavationNotification { copy(name = name) }
-            }
-        }
-    }
+    fun withName(name: String): HakemusBuilder =
```

Comment:
While rebasing, I refactored these to make it easier to add more modifier methods.

Reply:
Nice! How would the syntax be for something that is only on e.g. excavation notification?

Reply:
Either
```
    fun withInvoicingCustomer(invoicingCustomer: InvoicingCustomer) =
        updateApplicationData(
            { this },
            { copy(invoicingCustomer = invoicingCustomer) }
        )
```
Or
```
    fun withInvoicingCustomer(invoicingCustomer: InvoicingCustomer) =
        updateApplicationData(
            { throw UnsupportedOperationException() },
            { copy(invoicingCustomer = invoicingCustomer) }
        )
```
Or use something completely different, like the `onExcavationNotification()` removed here.